packages/runtime-core/src/apiWatch.ts

```diff
@@ -360,7 +360,10 @@ function doWatch(
   } else {
     // default: 'pre'
     job.pre = true
-    if (instance) job.id = instance.uid
+    if (instance) {
```

Comment:
this change is unnecessary.
we just check `cb.id !== instance.uid` in flushPreFlushCbs will be ok.

Reply:
Thanks. Updated.